Index: jessie-stage2/rootfs/etc/apt/apt.conf.d/50unattended-upgrades
===================================================================
--- jessie-stage2.orig/rootfs/etc/apt/apt.conf.d/50unattended-upgrades
+++ jessie-stage2/rootfs/etc/apt/apt.conf.d/50unattended-upgrades
@@ -24,7 +24,7 @@ Unattended-Upgrade::Origins-Pattern {
         // Codename based matching:
         // This will follow the migration of a release through different
         // archives (e.g. from testing to stable and later oldstable).
-//      "o=Raspbian,n=jessie";
+      "o=*,n=jessie";
 
         // Archive or Suite based matching:
         // Note that this will silently match a different release after
@@ -80,7 +80,7 @@ Unattended-Upgrade::Package-Blacklist {
 // If automatic reboot is enabled and needed, reboot at the specific
 // time instead of immediately
 //  Default: "now"
-//Unattended-Upgrade::Automatic-Reboot-Time "02:00";
+Unattended-Upgrade::Automatic-Reboot-Time "00:00";
 
 // Use apt bandwidth limit feature, this example limits the download
 // speed to 70kb/sec
