Index: patch/init_resize.sh
===================================================================
--- jessie-stage2.orig/rootfs/usr/lib/raspi-config/init_resize.sh
+++ jessie-stage2/rootfs/usr/lib/raspi-config/init_resize.sh
@@ -175,6 +175,13 @@ sed -i 's| init=/usr/lib/raspi-config/in
 if ! grep -q splash /boot/cmdline.txt; then
   sed -i "s/ quiet//g" /boot/cmdline.txt
 fi
+
+# hack to modify hostname in EthRaspbian
+MAC_HASH=`cat /sys/class/net/eth0/address | sha256sum | awk '{print substr($0,0,9)}'`
+echo ethnode-$MAC_HASH > /etc/hostname
+sed -i "s/127.0.1.1.*/127.0.1.1\tethnode-$MAC_HASH/g" /etc/hosts
+
+
 sync
 
 echo 1 > /proc/sys/kernel/sysrq
